File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        if(s1.length()>s2.length()) return false;
        int[] frqS1=new int[26];
        int[] frqS2=new int[26];
        for(int i=0;i<s1.length();i++) frqS1[s1.charAt(i)-'a']++;
        for(int i=0;i<s1.length();i++) frqS2[s2.charAt(i)-'a']++;
        for(int i=s1.length();i<s2.length();i++){
            if(Arrays.equals(frqS1,frqS2)) return true;
            frqS2[s2.charAt(i-s1.length())-'a']--;
            frqS2[s2.charAt(i)-'a']++;
        }
        return Arrays.equals(frqS1,frqS2);
    }
}

File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if(s2.size()<s1.size()) return false;
        vector<int> frqS1(26,0),frqS2(26,0);
        for(int i=0;i<s1.size();i++) frqS1[s1[i]-'a']++;
        for(int i=0;i<s1.size();i++) frqS2[s2[i]-'a']++;
        for(int i=s1.size();i<s2.size();i++){
            if(frqS1==frqS2) return true;
            frqS2[s2[i]-'a']++;
            frqS2[s2[i-s1.size()]-'a']--;
        }
        return frqS1==frqS2;
    }
};